src/main/java/io/vertx/ext/shell/command/base/VerticleDeploy.java

```diff
@@ -50,25 +52,45 @@
 public class VerticleDeploy extends AnnotatedCommand {
 
   private String name;
+  private String options;
 
   @Argument(index = 0, argName = "name")
   @Description("the verticle name")
   public void setName(String name) {
     this.name = name;
   }
 
+  @Argument(index = 1, argName = "options", required = false)
+  @Description("the verticle deployment options as JSON string")
+  public void setOptions(String options) {
+    this.options = options;
+  }
+
+  protected DeploymentOptions getDeploymentOptions() {
+    return (options != null && !options.isEmpty()) ?
+      new DeploymentOptions(new JsonObject(options)) : new DeploymentOptions();
+  }
+
   @Override
   public void process(CommandProcess process) {
-    process.vertx().deployVerticle(name, ar -> {
-      if (ar.succeeded()) {
-        process.write("Deployed " + ar.result() + "\n").end();
-      } else {
-        process.write("Could not deploy " + name + "\n");
-        StringWriter buffer = new StringWriter();
-        PrintWriter writer = new PrintWriter(buffer);
-        ar.cause().printStackTrace(writer);
-        process.write(buffer.toString()).end();
-      }
-    });
+    try {
+      process.vertx().deployVerticle(name, getDeploymentOptions(), ar -> {
+        if (ar.succeeded()) {
+          process.write("Deployed " + ar.result() + "\n").end();
+        } else {
+          process.write("Could not deploy " + name + "\n");
+          StringWriter buffer = new StringWriter();
+          PrintWriter writer = new PrintWriter(buffer);
+          ar.cause().printStackTrace(writer);
+          process.write(buffer.toString()).end();
+        }
+      });
+    } catch (RuntimeException e) {
```

Comment:
it would be more readable and better to have only the getDeploymentOptions() in the try block and not the whole block that deploys the verticle